cmd/endpointQuerier/main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"context"
+	"encoding/xml"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptrace"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"../../internal/endpoints"
+	"../../internal/fhir"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Metrics collected inside build-in prometheus vector. Each METRIC has its own registrations
+var httpCodesGaugeVec *prometheus.GaugeVec
+var responseTimeGaugeVec *prometheus.GaugeVec
+var tlsVersionGaugeVec *prometheus.GaugeVec
+var fhirVersionGaugeVec *prometheus.GaugeVec
+var totalUptimeChecksCounterVec *prometheus.CounterVec
+var totalFailedUptimeChecksCounterVec *prometheus.CounterVec
+
+// Need to define timeout or else it is infinite
+var netClient = &http.Client{
+	Timeout: time.Second * 35,
+}
+
+func getHTTPRequestTimingFor(url string, organizationName string, recordLongRunning bool) {
+	// Specifically query the FHIR endpoint metadata
+	url += "metadata"
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		println("HTTP Request Error: %s", err)
+	}
+
+	var start time.Time
+	trace := &httptrace.ClientTrace{}
+
+	// Drop connection if no reply within 30 seconds
+	ctx, cancel := context.WithDeadline(req.Context(), time.Now().Add(30*time.Second))
```

Comment:
I guess they _could_ be the same value, the difference is that the 30 seconds is what is defined to be the timeout for our particular HTTP request in this context. The 35 seconds is part of a httpClient best-practice since it does not define a timeout by default. See more information about the missing default timeout here https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779